File: evidence/generated-systems/angular/manufacturing-on-angular/src/app/models/Asset.js

// Define collection and schema for Asset
export interface Asset {
    assetTag:
	type : string
    assetName:
	type : string
    commissioningDate:
	type : Date
    Plant:
	type : Schema.Types.ObjectId
    WorkCenter:
	type : Schema.Types.ObjectId
    MaintenanceOrders:
 	type : [{ type: Schema.Types.ObjectId, ref: 'MaintenanceOrder' }]
    MaintenancePlans:
 	type : [{ type: Schema.Types.ObjectId, ref: 'MaintenancePlan' }]
    AssetStatus:
 	type : String
#
    collection: 'assets'
}

File: evidence/generated-systems/django/manufacturing-on-django/manufacturingOnDjango/models/Asset.py
from django.db import models
from manufacturingOnDjango.models.AssetStatus import AssetStatus

#======================================================================
# 
# Encapsulates data for model Asset
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class Asset Declaration
#======================================================================
class Asset (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	assetTag = models.CharField(max_length=200, null=True)
	assetName = models.CharField(max_length=200, null=True)
	commissioningDate = models.DateField(null=True)
	plant = models.ForeignKey('Plant', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	workCenter = models.ForeignKey('WorkCenter', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	maintenanceOrders = models.ManyToManyField('MaintenanceOrder',  blank=True, related_name='+')
	maintenancePlans = models.ManyToManyField('MaintenancePlan',  blank=True, related_name='+')
	assetStatus = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in AssetStatus])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.assetTag
		str = str + self.assetName
		str = str + self.commissioningDate
		str = str + self.assetStatus
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "Asset";
    
	def objectType(self):
		return "Asset";
